train.py

```diff
@@ -115,150 +109,166 @@ def train(model, dataset, data_csv, binary, params_file, augmentation, save, ver
-        training_accuracies.append((100. * train_correct / total))
+                # For binary classification, transform labels to one-vs-rest
+                if binary:
+                    target = target.eq(3).type(torch.int64)
 
-    # Calculate train loss and accuracy as an average of the last min(5, N_EPOCHS) losses or accuracies
-    train_loss = statistics.mean(training_losses[min(-N_EPOCHS, -5):])
-    train_accuracy = statistics.mean(training_accuracies[min(-N_EPOCHS, -5):])
+                output = model(data)
+                test_loss += loss_function(output, target).item()
 
-    logger.info("Final training score: Loss: %.4f, Accuracy: %.3f%%" % (train_loss, train_accuracy))
+                pred = output.max(1, keepdim=True)[1]
 
-    plt.plot(range(N_EPOCHS), training_losses, label = "Training loss")
-    plt.xlabel('epoch')
-    plt.ylabel('loss')
-    plt.title('Loss')
-    plt.legend()
-    plt.show()
+                correct = pred.eq(target.view_as(pred)).sum().item()
+                test_correct += correct
+                total += data.shape[0]
 
-    plt.plot(range(N_EPOCHS), training_accuracies, label = "Training accuracy")
-    plt.xlabel('epoch')
-    plt.ylabel('accuracy')
-    plt.title('Accuracy')
-    plt.legend()
-    plt.show()
+                test_loss /= len(test_plant_dataloader.dataset)
 
-    # %%
+                pred_list = torch.flatten(pred).cpu().numpy()
+                y_pred.extend(pred_list)
 
-    # test
-    test_loss = 0
-    test_correct = 0
-    total = 0
-    y_pred = []
-    y_true = []
+                target_list = target.cpu().numpy()
+                y_true.extend(target_list)
 
-    logger.info("Starting testing cycle")
+        test_accuracy = 100. * test_correct / total
 
-    with torch.no_grad():
-        for batch_num, batch in enumerate(test_plant_dataloader):
-            data, target = batch['image'].to(device), batch['label'].to(device)
+        logger.info("Final test score: Loss: %.4f, Accuracy: %.3f%%" % (test_loss, test_accuracy))
 
-            # For binary classification, transform labels to one-vs-rest
-            if binary:
-                target = target.eq(3).type(torch.int64)
-
-            output = model_class(data)
-            test_loss += loss_function(output, target).item()
-
-            pred = output.max(1, keepdim=True)[1]
-
-            correct = pred.eq(target.view_as(pred)).sum().item()
-            test_correct += correct
-            total += data.shape[0]
-
-            test_loss /= len(test_plant_dataloader.dataset)
-
-            pred_list = torch.flatten(pred).cpu().numpy()
-            y_pred.extend(pred_list)
-            
-            target_list = target.cpu().numpy()
-            y_true.extend(target_list)
-
-    test_accuracy = 100. * test_correct / total
-
-    logger.info("Final test score: Loss: %.4f, Accuracy: %.3f%%" % (test_loss, test_accuracy))
+        other_json = None
 
     if binary:
         labels = ['Non-VD', 'VD']
```

Comment:
Maybe we could likewise put all this to a function train_nn

Reply:
That would improve the code readability, but I don't know if we want to do more refactoring anymore since the project is ending (and the current solution works too :D)